components/formats-bsd/src/loci/formats/in/APNGReader.java

```diff
@@ -420,12 +437,168 @@ else if (compression == 0) {
           int passWidth = PASS_WIDTHS[i] * nColBlocks;
           int passHeight = PASS_HEIGHTS[i] * nRowBlocks;
 
+          // see http://www.libpng.org/pub/png/spec/1.2/PNG-DataRep.html#DR.Interlaced-data-order
+          // for clarification of the pass width / pass row adjustments
+
+          if (nColBlocks * 8 != width) {
+            int extraCols = getSizeX() - (nColBlocks - 1) * 8;
+            switch (extraCols) {
+              case 1:
+                if (i == 1 || i == 3 || i == 5) {
+                  passWidth -= PASS_WIDTHS[i];
+                }
+                if (i == 2 || i == 4 || i == 6) {
+                  passWidth -= (PASS_WIDTHS[i] - 1);
+                }
+                break;
+              case 2:
+                if (i == 1 || i == 3) {
+                  passWidth -= PASS_WIDTHS[i];
+                }
+                if (i == 2 || i == 4 || i == 5) {
+                  passWidth -= (PASS_WIDTHS[i] - 1);
+                }
+                if (i == 6) {
+                  passWidth -= (PASS_WIDTHS[i] - 2);
+                }
+                break;
+              case 3:
+                if (i == 1) {
+                  passWidth -= PASS_WIDTHS[i];
+                }
+                if (i == 2 || i == 3 || i == 5) {
+                  passWidth -= (PASS_WIDTHS[i] - 1);
+                }
+                if (i == 4) {
+                  passWidth -= (PASS_WIDTHS[i] - 2);
+                }
+                if (i == 6) {
+                  passWidth -= (PASS_WIDTHS[i] - 3);
+                }
+                break;
+              case 4:
+                if (i == 1) {
+                  passWidth -= PASS_WIDTHS[i];
+                }
+                if (i == 2 || i == 3) {
+                  passWidth -= (PASS_WIDTHS[i] - 1);
+                }
+                if (i == 4 || i == 5) {
+                  passWidth -= (PASS_WIDTHS[i] - 2);
+                }
+                if (i == 6) {
+                  passWidth -= (PASS_WIDTHS[i] - 4);
+                }
+                break;
+              case 5:
+                if (i == 3) {
+                  passWidth -= (PASS_WIDTHS[i] - 1);
+                }
+                if (i == 5) {
+                  passWidth -= (PASS_WIDTHS[i] - 2);
+                }
+                if (i == 4) {
+                  passWidth -= (PASS_WIDTHS[i] =- 3);
```

Comment:
is this second `=` desired? (also could make these "else if"'s but doesn't much matter really)